rapids-cmake/find/package.cmake

```
@@ -75,7 +62,39 @@ so users have consistency. List all targets used by your project in `GLOBAL_TARG
 
 ``FIND_ARGS``
   Required placeholder to be provied before any extra arguments that need to
-  be passed down to cmake:command:`find_pacakge`
+  be passed down to cmake:command:`find_pacakge <cmake:command:find_package>`
+
+.. note::
+  If the project/package you are looking for doesn't have an existing
+  CMake Find module, please look at using :cmake:command:`rapids_find_generate_module`.
+
+The :cmake:command:`rapids_find_package` function supports two call modes.
+
+  1. Is when all the parameters for :cmake:command:`find_package <cmake:command:find_package>`
+  are first followed by rapids parameters such as `BUILD_EXPORT_SET` and
+  `INSTALL_EXPORT_SET` last. Here is an example on what this call would look like:
+
+  .. code-block:: cmake
+
+    rapids_find_package(ZLIB 1.2 REQUIRED
+      GLOBAL_TARGETS ZLIB::ZLIB
+      INSTALL_EXPORT_SET my-export-set
+      BUILD_EXPORT_SET my-export-set
+    )
+
+  2. Is when the rapids parameters come first, and in that case they must be
+  preceded by the `FIND_ARGS` keyword. This ensures proper argument propagation
+  to the underlying :cmake:command:`find_package <cmake:command:find_package>`.
+  Here is an example on what this call would look like:
+
+  .. code-block:: cmake
+
+    rapids_find_package(ZLIB
+      GLOBAL_TARGETS ZLIB::ZLIB
+      INSTALL_EXPORT_SET my-export-set
+      BUILD_EXPORT_SET my-export-set
+      FIND_ARGS 1.2 QUIET REQUIRED
```

I assume that there's no particular reason to include QUIET here. I would remove it, or alternatively also add it to the above example, so that there's no differences that aren't directly relevant, otherwise it might lead to confusion.
```suggestion
      FIND_ARGS 1.2 REQUIRED
```

---

Yeah lets keep the symetry between the examples.